shared/common-adapters/name-with-icon/container.js

```diff
@@ -0,0 +1,61 @@
+// @flow
+import {compose, connect, setDisplayName} from '../../util/container'
+import * as Route from '../../actions/route-tree'
+import {teamsTab} from '../../constants/tabs'
+import * as ProfileGen from '../../actions/profile-gen'
+import * as TrackerGen from '../../actions/tracker-gen'
+import NameWithIcon, {type NameWithIconProps} from '.'
+import {isMobile} from '../../constants/platform'
+
+export type ConnectedNameWithIconProps = {|
+  ...NameWithIconProps,
+  onClick?: ((SyntheticEvent<> | void) => void) | 'tracker' | 'profile' | 'trackerProfileFallback',
+|}
+
+const mapStateToProps = () => ({})
+
+const mapDispatchToProps = dispatch => ({
+  onOpenTeamProfile: (teamname: string) =>
+    dispatch(Route.navigateTo([teamsTab, {props: {teamname: teamname}, selected: 'team'}])),
+  onOpenTracker: (username: string) =>
+    dispatch(TrackerGen.createGetProfile({forceDisplay: true, ignoreCache: true, username})),
+  onOpenUserProfile: (username: string) => dispatch(ProfileGen.createShowUserProfile({username})),
+})
+
+const mergeProps = (stateProps, dispatchProps, ownProps: ConnectedNameWithIconProps) => {
+  const {onClick, ...props} = ownProps
+
+  let functionOnClick
+  let clickType
+  if (!isMobile && (onClick === 'tracker' || onClick === 'trackerProfileFallback')) {
+    functionOnClick = () => {
```

Comment:
perhaps:

```
if (ownProps.username) {
  functionOnClick = () => dispatchProps.onOpenTracker(ownProps.username)
}
```
?

Reply:
Huh this was my first thought but it resulted in some weird flow errors. Seems to be working now though so I guess those were from something else. 🤷‍♂️